### PR TITLE
Ensure its possible to `remove` stack with missing bucket

### PR DESCRIPTION
**Pull request by pgrzesik**
Addresses the `remove` part of https://github.com/serverless/serverless/issues/3964

I've also cleaned up the tests in the process because there were a few leftover tests in `bucket.test.js` 

Addressing `deploy` scenario will come in a separate PR

### CODE REVIEW

File: lib/plugins/aws/remove/lib/bucket.js [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
+const { legacy, log } = require('@serverless/utils/log');
 
 module.exports = {
   async setServerlessDeploymentBucketName() {
-    return this.provider.getServerlessDeploymentBucketName().then((bucketName) => {
-      this.bucketName = bucketName;
-    });
+    const bucketName = await this.provider.getServerlessDeploymentBucketName();
+    this.bucketName = bucketName;
+    this.checkIfBucketExists(bucketName);
```
Comment (medikoo) [inline lib/plugins/aws/remove/lib/bucket.js:9]: What's the purpose of that call?

File: lib/plugins/aws/remove/lib/bucket.js [hunk 2]
```diff
@@ -75,21 +71,23 @@ module.exports = {
...
-    return BbPromise.resolve();
   },
 
   async emptyS3Bucket() {
-    return BbPromise.bind(this)
-      .then(this.setServerlessDeploymentBucketName)
-      .then(this.listObjects)
-      .then(this.deleteObjects);
+    await this.setServerlessDeploymentBucketName();
+    if (this.bucketName && (await this.checkIfBucketExists(this.bucketName))) {
```
Comment (medikoo) [inline lib/plugins/aws/remove/lib/bucket.js:85]: Wouldn't `bucketName` be unconditionally set here? (we didn't have that check before)

File: lib/plugins/aws/remove/lib/bucket.js (hunk 1 above)
Comment (pgrzesik) [reply lib/plugins/aws/remove/lib/bucket.js]: While implementing the second PR I've discovered a bug in this implementation for a specific case and I've adjusted the implementation a little bit

File: lib/plugins/aws/remove/lib/bucket.js (hunk 2 above)
Comment (pgrzesik) [reply lib/plugins/aws/remove/lib/bucket.js:95]: There is a scenario in which it won't be set so we need to leave this check this way

File: lib/plugins/aws/remove/lib/bucket.js (hunk 1 above)
Comment (medikoo) [reply lib/plugins/aws/remove/lib/bucket.js]: Ok, but we do not take result of that method. What is the reason of checking if it exists, if we're not interested in result?

File: lib/plugins/aws/remove/lib/bucket.js (hunk 2 above)
Comment (medikoo) [reply lib/plugins/aws/remove/lib/bucket.js:95]: Is this a scenario that was already in `master` and was not handled? Or is this scenario introduced with this PR?

Also when exactly is that the case?

Comment (pgrzesik) [reply lib/plugins/aws/remove/lib/bucket.js:95]: Yes, in `master` it was just throwing an error on SDK call in `setServerlessDeploymentBucket`, now we want to handle such situation gracefully

File: lib/plugins/aws/remove/lib/bucket.js (hunk 1 above)
Comment (pgrzesik) [reply lib/plugins/aws/remove/lib/bucket.js]: This code has been changed, please see the updated version

**Review by medikoo — APPROVED** [review]
Looks great 👍

**Pull request merged**